### ISSUE TITLE
Test response view data doesn't contain shared props

### ISSUE THREAD

**Issue by chargefield** [inertiajs/inertia-laravel#32]
I'm trying to test a route that should pass the auth user as a shared prop data, along with others(app.name...etc). I am only seeing the prop data defined in the controller.

Anyone run into this issue? Any suggestions... @reinink 

btw, all shared prop data are being passed down to the Vue page component successfully and the json_encoded version can be seen in page view source. I just can't test it.

**Comment by reinink** [comment]
Have you reviewed the tests in the PingCRM app? See [here](https://github.com/inertiajs/pingcrm/tree/master/tests). Might be helpful. In particular, see the test macros [here](https://github.com/inertiajs/pingcrm/blob/master/tests/TestCase.php#L18-L52).

**Comment by chargefield** [comment]
I have copied those macros already :)

When i call $this->assertHasProp('auth') it doesn't exist. Even $this->assertHasProp('app.name')...

$this->original->getData() only outputs the data provided to the Inertia::render function. I'm not getting any of the shared data.

Only in my tests...

**Comment by georgehanson** [comment]
Just in addition to this, it might be useful to have those assertions on a trait which can be applied to a Laravel test suite - rather than having to create the macros yourself.

What do you think @reinink?

**Comment by georgehanson** [comment]
Ignore that, I've just noticed #51 :)

**Comment by rakk7** [comment]
Yeah, same issue here.. 

When calling ` $this->assertHasProp('auth')` it doesn't exist.. the response doesn't contain the `shared data`.

But when I open some URL in the browser the "auth" prop does exist!

**Comment by Juhlinus** [comment]
Hello, @rakk7 !

Those methods does not exist outside of the PingCRM as of right now. If you want to use them then check out the macros:

https://github.com/inertiajs/pingcrm/blob/master/tests/TestCase.php#L18-L52

Or review my PR: #51

**Comment by rakk7** [comment]
@Juhlinus 

Yeah, I already have these methods. I am facing the same issue as @chargefield.

**Comment by mattharcourt** [comment]
I just ran into this myself... 

Not sure why, but in the test environment (not in the browser) it appears that the Inertia Facade returns a different instance of the `Inertia\ResponseFactory` when calling `Inertia::render()` than was returned when `Inertia::share()` was called when the AppServiceProvider was registered. As a result, the ResponseFactory used by the render method didn't have any shared props. 

I was able to solve when I changed the Inertia Facade to always return the same instance by binding the ResponseFactory class into the app container as a singleton in the `Inertia\ServiceProvider` register method.

This has been proposed to solve a different issue (#73)  by @rojtjo in PR #75
